chainer/distributions/bernoulli.py

```diff
@@ -0,0 +1,119 @@
+import chainer
+from chainer.backends import cuda
+from chainer import distribution
+from chainer.functions.activation import sigmoid
+from chainer.functions.array import broadcast
+from chainer.functions.math import exponential
+import numpy
+
+
+class Bernoulli(distribution.Distribution):
+
+    """Bernoulli Distribution.
+
+    The probability mass function of the distribution is expressed as
+
+    .. math::
+        P(x = 1; p) = p \\\\
+        P(x = 0; p) = 1 - p
+
+    Args:
+        p(:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+        :class:`cupy.ndarray`): Parameter of distribution representing \
+        :math:`p`. Either `p` or `logit` (not both) must have a value.
+        logit(:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+        :class:`cupy.ndarray`): Parameter of distribution representing \
+        :math:`\\log\\{p/(1-p)\\}`. Either `p` or `logit` (not both) must \
+        have a value.
+
+    """
+
+    def __init__(self, p=None, logit=None):
+        super(Bernoulli, self).__init__()
+        if not (p is None) ^ (logit is None):
+            raise ValueError(
+                "Either `p` or `logit` (not both) must have a value.")
+
+        with chainer.using_config('enable_backprop', True):
+            if p is None:
+                self.logit = chainer.as_variable(logit)
+                self.p = sigmoid.sigmoid(self.logit)
+            else:
+                self.p = chainer.as_variable(p)
+                self.logit = exponential.log(self.p) - \
+                    exponential.log(1. - self.p)
+
+    @property
+    def batch_shape(self):
+        return self.p.shape
+
+    @property
+    def entropy(self):
+        return -self.p * exponential.log(self.p) - \
+            (numpy.float32(1.) - self.p) \
+            * exponential.log(numpy.float32(1.) - self.p)
+
+    @property
+    def event_shape(self):
+        return ()
+
+    @property
+    def _is_gpu(self):
+        return isinstance(self.p.data, cuda.ndarray)
+
+    def log_prob(self, x):
+        if type(x) == chainer.Variable:
+            x = x.data
+        x = x.astype(self.p.dtype)
+        if self._is_gpu:
+            invalid_inf = cuda.cupy.zeros_like(x)
+            invalid = cuda.cupy.bitwise_and(x != 0, x != 1)
```

Comment:
Why do you want to validate `x`?  Do you expect other integers, or any floats?

Reply:
Yes. When x is not 0 or 1, this function returns -inf.

Reply:
Um.., I also think that it is better to support an option not to validate as like `F.bernoulli_nll`.